### DISCUSSION TITLE
Backup to Multiple Archives in Different Locations

### DISCUSSION THREAD

**Discussion by ntrkd** [nextcloud/all-in-one#4181] [Ideas] [answered]
### Is your feature request related to a problem? Please describe. 
Not necessarily

### Describe the solution you'd like
Add the ability for the default AIO backup solution to backup to multiple locations **using multiple borg archives**

### Describe alternatives you've considered
The [AIO documentation](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#sync-the-backup-regularly-to-another-drive) suggests using rsync to copy the archive to multiple locations. However, [borg backup](https://borgbackup.readthedocs.io/en/stable/faq.html#can-i-copy-or-synchronize-my-repo-to-another-location) (the tool used by AIO backup) specifically does _not_ recommend this. Instead they recommend using multiple borg archives to prevent errors in one of them occurring in all.

### Additional context
The Backup app, available in nextcloud has the ability to backup to multiple locations and it checks for damages to the archives, but afaik they also use rsync. Please let me know if you need any more information.

**Comment by szaimen** [comment]
Hi, thanks for you idea! Unfortunately as this sounds like a nice feature, the requests for this are quite low. Currently there are no plans to implement such a feature.

However as a workaround you might simply check the backup integrity using borg check before doing the syncing with rsync which should be pretty reliable.

**Reply by ntrkd** [reply]
Thanks for an answer! I think I might just have to do that. But on a side note, it might be useful to have pre and post hooks for the backup. For example, I have a HDD that is used exclusively for backup. I use hdparm -Y to power it down after backups, so having a pre hook to wake it up then a post hook to power it down would be nice. Might also be easier to rsync using this method.

**Reply by ntrkd** [reply]
Also for other's that might stumble upon this thread, I'm planning on using the backup app from nextcloud which does support multiple backup locations (it will use borg check automatically and fix errors) in combination with the AIO backup. So following the 3-2-1 method, 1 backup from AIO backup and 1 or 2 using the backup app for a total of 2-3 backups depending if you consider your production copy a backup. This would mean we do have at least 2 borg archives to prevent errors.

**Comment by ntrkd** [comment]
There's another way you can do it besides using the native backup solutions. Most of the data is just stored in volumes (besides your data dir). So you can manually back them up or write a script to do so. Your volumes are most likely stored in /var/lib/docker/volumes so just dump the database using pg_dump (if using postgreSQL) and use borg/restic to backup those volumes.

Here's a really nice guide -> [rair.dev](https://rair.dev/nextcloud-backup-pt-1/#1-restic-)

**Reply by szaimen** [reply]
We have this documentation btw: https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by ntrkd** [reply]
Additionally, if you plan on storing your backup only on a remote cloud provider without syncing a directory on your machine (only keeping remote copy), use restic instead since it supports some providers and supports rclone which in turn supports a wide variety. Borg does not work with rclone

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/discussions/7009#discussioncomment-14727561
